app/controllers/spree/api/v2/storefront/reviews_controller.rb

```diff
@@ -0,0 +1,97 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class ReviewsController < ::Spree::Api::V2::BaseController
+          include Spree::Api::V2::CollectionOptionsHelpers
+
+          before_action :load_product, only: [:index, :create]
+
+          def index
+            render_serialized_payload {serialize_collection(paginated_collection)}
+          end
+
+          def show
+            render_serialized_payload {serialize_resource(resource)}
+          end
+
+          def create
+            # TODO move to service
+            # result = create_service.call(user: spree_current_user, review_params: review_params, product: @product, ip_address: request.remote_ip)
+
+            params[:review][:rating].sub!(/\s*[^0-9]*\z/, '') unless params[:review][:rating].blank?
+
+            @review = Spree::Review.new(review_params)
+            @review.product = @product
+            @review.user = spree_current_user if spree_user_signed_in?
```

Comment:
TODO: @poudelprakash code was copied from backend, ensure current user is fetched properly